src/platform/ESP32/ESP32Utils.cpp

```diff
@@ -237,3 +238,70 @@ bool ESP32Utils::HasIPv6LinkLocalAddress(tcpip_adapter_if_t intfId)
     ip6_addr_t unused;
     return tcpip_adapter_get_ip6_linklocal(intfId, &unused) == ESP_OK;
 }
+
+CHIP_ERROR ESP32Utils::GetWiFiStationProvision(Internal::DeviceNetworkInfo & netInfo, bool includeCredentials)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    wifi_config_t stationConfig;
+
+    err = esp_wifi_get_config(ESP_IF_WIFI_STA, &stationConfig);
+    SuccessOrExit(err);
+
+    VerifyOrExit(stationConfig.sta.ssid[0] != 0, err = CHIP_ERROR_INCORRECT_STATE);
+
+    netInfo.NetworkId              = kWiFiStationNetworkId;
+    netInfo.FieldPresent.NetworkId = true;
+    memcpy(netInfo.WiFiSSID, stationConfig.sta.ssid, min(strlen((char *) stationConfig.sta.ssid) + 1, sizeof(netInfo.WiFiSSID)));
```

Comment:
nit: should we use C++-style cast here? also should it be const char * ?

Should we ensure a null terminator even when ssid length overflows and sizeof is used?

Reply:
+1

Reply:
@shana-apple ?